internal/scheduler/simulator/simulator_test.go

```diff
@@ -297,6 +299,149 @@ func TestSimulator(t *testing.T) {
+							{
+								Number:            32,
+								JobSet:            "job-set-1",
+								PriorityClassName: "armada-preemptible-away",
+								Requirements: schedulerobjects.PodRequirements{
+									ResourceRequirements: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											"cpu":            resource.MustParse("16"),
+											"memory":         resource.MustParse("512Gi"),
+											"nvidia.com/gpu": resource.MustParse("1"),
+										},
+									},
+								},
+								RuntimeDistribution: ShiftedExponential{Minimum: time.Hour},
+							},
+						},
+					},
+				},
+			},
+			schedulingConfig: func() configuration.SchedulingConfig {
+				config := testfixtures.TestSchedulingConfig()
+				config.Preemption.PriorityClasses = map[string]types.PriorityClass{
+					"armada-preemptible-away": {
+						Priority:    30000,
+						Preemptible: true,
+
+						AwayNodeTypes: []types.AwayNodeType{{Priority: 29000, WellKnownNodeTypeName: "gpu-whale"}},
+					},
+					"armada-preemptible": {
+						Priority:    30000,
+						Preemptible: true,
+					},
+				}
+				config.Preemption.DefaultPriorityClass = "armada-preemptible"
+				config.WellKnownNodeTypes = []configuration.WellKnownNodeType{
+					{
+						Name:   "gpu-whale",
+						Taints: []v1.Taint{{Key: "gpu-whale", Value: "true", Effect: v1.TaintEffectNoSchedule}},
+					},
+				}
+				return config
+			}(),
+			expectedEventSequences: armadaslices.Concatenate(
+				[]*armadaevents.EventSequence{
+					{Queue: "queue-0", JobSetName: "job-set-0", Events: armadaslices.Repeat(1, SubmitJob())},
+					{Queue: "queue-1", JobSetName: "job-set-1", Events: armadaslices.Repeat(32, SubmitJob())},
+				},
+				armadaslices.Repeat(1, &armadaevents.EventSequence{Queue: "queue-0", JobSetName: "job-set-0", Events: []*armadaevents.EventSequence_Event{JobRunLeased()}}),
+				armadaslices.Repeat(24, &armadaevents.EventSequence{Queue: "queue-1", JobSetName: "job-set-1", Events: []*armadaevents.EventSequence_Event{JobRunLeased()}}),
+				armadaslices.Repeat(1, &armadaevents.EventSequence{Queue: "queue-0", JobSetName: "job-set-0", Events: []*armadaevents.EventSequence_Event{JobSucceeded()}}),
+				[]*armadaevents.EventSequence{
+					{Queue: "queue-0", JobSetName: "job-set-0", Events: armadaslices.Repeat(4, SubmitJob())},
+				},
+				armadaslices.Repeat(8, &armadaevents.EventSequence{Queue: "queue-1", JobSetName: "job-set-1", Events: []*armadaevents.EventSequence_Event{JobRunLeased()}}),
+				armadaslices.Repeat(24, &armadaevents.EventSequence{Queue: "queue-1", JobSetName: "job-set-1", Events: []*armadaevents.EventSequence_Event{JobRunPreempted()}}),
+				armadaslices.Repeat(3, &armadaevents.EventSequence{Queue: "queue-0", JobSetName: "job-set-0", Events: []*armadaevents.EventSequence_Event{JobRunLeased()}}),
+				armadaslices.Repeat(24, &armadaevents.EventSequence{Queue: "queue-1", JobSetName: "job-set-1", Events: []*armadaevents.EventSequence_Event{SubmitJob()}}),
```

Comment:
This sequence of events is hard to read, but it shows an interesting scheduling phenomenon:

In this test there are four nodes, let's call them A<sub>0</sub>, A<sub>1</sub>, B<sub>0</sub>, and B<sub>1</sub>, where B<sub>0</sub> and B<sub>1</sub> have the taint `gpu-whale=true:NoSchedule`. At the start of this scheduling round the nodes are occupied by 32 jobs from `queue-0`. The rest of the scheduling round proceeds like this:

1. The scheduler evicts all of the jobs from `queue-0`, because they are preemptible.
2. The scheduler schedules seven evicted jobs from `queue-0` back on A<sub>0</sub>.
3. The scheduler schedules one job from `queue-1` on A<sub>1</sub>.
4. The scheduler schedules the remaining evicted job from `queue-0` that was already running on A<sub>0</sub>.
5. None of the jobs from `queue-0` that we already running on A<sub>1</sub> can be scheduled anymore, because a job from `queue-1` has already been scheduled there at the same priority. None of the away jobs from `queue-0` that are running on B<sub>0</sub> or B<sub>1</sub> can be scheduled anymore, because home jobs from `queue-1` are preferred instead.

The outcome of this scheduling round is that 24 out of 32 running jobs from `queue-0` are preempted, and three jobs from `queue-1` are scheduled in their place.